core/providers/java/gradle.go

```diff
@@ -0,0 +1,92 @@
+package java
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/railwayapp/railpack/core/generate"
+)
+
+const (
+	DEFAULT_GRADLE_VERSION = "8"
+	GRADLE_CACHE_KEY       = "gradle"
+)
+
+func (p *JavaProvider) usesGradle(ctx *generate.GenerateContext) bool {
+	return ctx.App.HasMatch("gradlew")
+}
+
+func (p *JavaProvider) setGradleVersion(ctx *generate.GenerateContext) error {
+	miseStep := ctx.GetMiseStepBuilder()
+	gradle := miseStep.Default("gradle", DEFAULT_GRADLE_VERSION)
+
+	if envVersion, _ := ctx.Env.GetConfigVariable("GRADLE_VERSION"); envVersion != "" {
+		miseStep.Version(gradle, envVersion, "GRADLE_VERSION")
+	}
```

Comment:
The name is the second value returned in the tuple

```suggestion
	if envVersion, envName := ctx.Env.GetConfigVariable("GRADLE_VERSION"); envVersion != "" {
		miseStep.Version(gradle, envVersion, envName)
	}
```